library/src/main/java/com/getstream/sdk/chat/storage/Storage.java

```diff
@@ -1,578 +1,47 @@
-
-        // based on https://gist.github.com/cesarferreira/ef70baa8d64f9753b4da
-        new QueryTask(l).execute(queryID);
-
-        return channelStates;
-    }
-
-
-    public QueryChannelsQ selectQuery(String queryID) {
-        if (!enabled) return null;
-
-        return queryChannelsQDao.select(queryID);
-    }
-
-
-    public void insertChannels(List<Channel> channels) {
-        if (!enabled) return;
-
-        for (Channel c : channels) {
-            c.preStorage();
-            c.getLastState().preStorage();
-        }
-
-        new AsyncTask<Void, Void, Void>() {
-            @Override
-            protected Void doInBackground(Void... voids) {
-                Log.i(TAG, String.format("Inserted %d channels into offline storage", channels.size()));
-                channelsDao.insertChannels(channels);
-                return null;
-            }
-        }.execute();
-    }
-
-    @Transaction
-    private void insertChannelStateInTransaction(List<User> users, QueryChannelsQ query, List<Channel> channels, List<Message> messages) {
-        insertUsersUnique(users);
-        Log.i(TAG, String.format("Inserted %d channels, %d messages into offline storage for query with id %s", channels.size(), messages.size(), query.getId()));
-        channelsDao.insertChannels(channels);
-        queryChannelsQDao.insertQuery(query);
-        for (Message m : messages) {
-            m.preStorage();
-        }
-        messageDao.insertMessages(messages);
-    }
-
-    private void insertUsersUnique(List<User> users) {
-        HashMap<String, User> userMap = new HashMap<>();
-        for (User u : users) {
-            if (u == null) continue;
-            userMap.put(u.getId(), u);
-        }
-
-        Log.i(TAG, String.format("Inserted %d users into offline storage", userMap.values().size()));
 
-        usersDao.insertUsers(new ArrayList(userMap.values()));
-    }
+public interface Storage {
 
-    private void enrichUsers(List<Channel> channels) {
-        List<String> userIDs = new ArrayList<>();
-        // gather all the user ids
+    MutableLiveData<ChannelState> selectChannelState(String cid, OnQueryListener l);
 
-        for (Channel c : channels) {
+    MutableLiveData<List<ChannelState>> selectChannelStates(String queryID, Integer limit, OnQueryListener l);
 
-            // gather the users from members, read, last message and created by
-            userIDs.add(c.getCreatedByUserID());
-            for (Member m : c.getLastState().getMembers()) {
-                userIDs.add(m.getUserId());
-            }
-            for (ChannelUserRead r : c.getLastState().getReads()) {
-                userIDs.add(r.getUserId());
-            }
-            Message lastMessage = c.getLastState().computeLastMessage();
-            if (lastMessage != null) {
-                userIDs.add(lastMessage.getUser().getId());
-            }
-        }
+    QueryChannelsQ selectQuery(String queryID);
```

Comment:
Documentation should in interfaces.
That's the first place where developers would/can go, as implementation might be eventually hidden behind DI. Also if we or developers will have few implementations we should maintain contract in a single place - interface.

Reply:
Yeah, I understand it. But there was no documentation for methods, only for `deleteChannel`. which I recently added. I just copied that we have.

Reply:
Oh, ok!